lib/carthage_cache/application.rb

```diff
@@ -16,8 +16,13 @@ def initialize(project_path, verbose, config, repository: AWSRepository, termina
       @terminal = terminal.new(verbose)
       @archiver = Archiver.new
       @config = Configurator.new(@terminal, project_path, config).config
-      clazz = @config.read_only? ? HTTPRepository : repository
-      @repository = clazz.new(@config.bucket_name, @config.hash_object[:aws_s3_client_options])
+      if @config.local_only?
+          cacheDirectory = File.join(project_path, "Carthage", @config.local_mode)
```

Comment:
I think it would be better if we don't require that the cache path must be relative to the project path. This way we could share the cache across projects. Also I think that a default value should be provided and that it should be `~/Library/Cache/carthage_cache`. Please make sure to create the directory if it doesn't exist.